### PR TITLE
[Config] imaging and eeg config groups

### PR DESCRIPTION
**Pull request by laemtl**
- create imaging and eeg config groups
- add new config EEGS3DataPath
- Add default value to electrophysiology_uploader UploadDate

### CODE REVIEW

**Review by regisoc — COMMENTED** [review]
Tested. 

image

@laemtl Should we add these changes in `SQL/0000-00-03-ConfigTables.sql`?

Else LGTM.

File: SQL/0000-00-03-ConfigTables.sql [hunk 1]
```diff
@@ -159,6 +156,9 @@ INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType,
...
 INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'MriPythonConfigFile', 'Name of the Python MRI config file (stored in dicom-archive/.loris_mri/)', 1, 0, 'text', ID, 'Name of the Python MRI config file', 28 FROM ConfigSettings WHERE Name="imaging_pipeline";
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'MRICodePath', 'Path to directory where Loris-MRI (git) code is installed', 1, 0, 'text', ID, 'LORIS-MRI code', 6 FROM ConfigSettings WHERE Name="imaging_pipeline";
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'MRIUploadIncomingPath', 'Path to the upload directory for incoming MRI studies', 1, 0, 'text', ID, 'MRI Incoming Directory', 7 FROM ConfigSettings WHERE Name="imaging_pipeline";
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'MINCToolsPath', 'Path to the MINC tools', 1, 0, 'web_path', ID, 'Path to the MINC tools', 12 FROM ConfigSettings WHERE Name="imaging_pipeline";
```
Comment (regisoc) [inline SQL/0000-00-03-ConfigTables.sql:161]: Is order important here? 
image

File: SQL/New_patches/2023-05-01-imaging-eeg-configs.sql [hunk 2]
```diff
@@ -0,0 +1,55 @@
+-- Add the EEG Pipeline Config group
```
[the remark this answers is not in the document]
Comment (laemtl) [reply SQL/New_patches/2023-05-01-imaging-eeg-configs.sql:1]: Screenshot 2023-12-12 at 11 27 29 AM

[the remark this answers is not in the document]
Comment (laemtl) [reply SQL/New_patches/2023-05-01-imaging-eeg-configs.sql:1]: I think it was :)

File: SQL/0000-00-03-ConfigTables.sql (hunk 1 above)
Comment (regisoc) [reply SQL/0000-00-03-ConfigTables.sql:161]: After discussion, looks like it is not.

**Review by regisoc — APPROVED** [review]

**Comment by driusan** [comment]
Tests are failing. Looks like there might be an SQL error introduced in the default schema?

File: SQL/0000-00-03-ConfigTables.sql [hunk 3]
```diff
@@ -69,13 +69,10 @@ INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType,
...
-INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'MINCToolsPath', 'Path to the MINC tools', 1, 0, 'web_path', ID, 'Path to the MINC tools', 12 FROM ConfigSettings WHERE Name="paths";
 INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'documentRepositoryPath', 'Path to uploaded document repository files', 1, 0, 'web_path', ID, 'Document Repository Upload Path', 13 FROM ConfigSettings WHERE Name="paths";
 INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'dataReleasePath', 'Path to uploaded data release files', 1, 0, 'web_path', ID, 'Data release Upload Path', 14 FROM ConfigSettings WHERE Name="paths";
 INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'EEGUploadIncomingPath', 'Path to the upload directory for incoming EEG studies', 1, 0, 'text', ID, 'EEG Incoming Directory', 7 FROM ConfigSettings WHERE Name="paths";
```
Comment (regisoc) [inline SQL/0000-00-03-ConfigTables.sql:78]: duplicated line

File: SQL/0000-00-03-ConfigTables.sql [hunk 4]
```diff
@@ -177,6 +177,10 @@ INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType,
...
 INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'AWS_S3_Default_Bucket', 'Default bucket for LORIS to use for accessing files in S3.', 1, 0, 'text', ID, 'AWS S3 Default Bucket', 3 FROM ConfigSettings WHERE Name='aws';
 
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, Label, OrderNumber) VALUES ('eeg_pipeline', 'EEG Pipeline settings', 1, 0, 'EEG Pipeline', 15);
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'EEGS3DataPath', 'EEG S3 data path for assembly data', 1, 0, 'text', ID, 'EEG S3 data path', 15 FROM ConfigSettings WHERE Name = 'eeg_pipeline';
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'EEGUploadIncomingPath', 'Path to the upload directory for incoming EEG studies', 1, 0, 'text', ID, 'EEG Incoming Directory', 7 FROM ConfigSettings WHERE Name="eeg_pipeline";
```
Comment (regisoc) [inline SQL/0000-00-03-ConfigTables.sql:182]: duplicated line

**Comment by regisoc** [comment]
Duplicated lines L78 and L182 in configTables.sql

**Review by regisoc — CHANGES_REQUESTED** [review]
Duplicated lines L78 and L182 in configTables.sql

**Review by regisoc — APPROVED** [review]

**Pull request merged**